test/e2e/framework/pod/utils.go

```diff
@@ -113,3 +114,19 @@ func GetLinuxLabel() *v1.SELinuxOptions {
 	return &v1.SELinuxOptions{
 		Level: "s0:c0,c1"}
 }
+
+// GetRestrictedPodSecurityContext returns a minimal restricted pod security context.
+func GetRestrictedPodSecurityContext() *v1.PodSecurityContext {
```

Comment:
Another useful function might be `MixinRestrictedSecurityContext(podSpec *v1.PodSpec) (*v1.PodSpec, error)`, which would set the PodSecurityContext, and a ContainerSecurityContext on each container. Optionally, if a security context already exists it could just fail, or it could attempt to merge the contexts.

Reply:
great idea, let me experiment with the implementation 👍 If possible, I'd like to contribute this as a follow-up.